### PR TITLE
fix(exec): add support for ImageMagick 7.0.1-Q16

### PR DESCRIPTION
**Pull request by RaschidJFR**
Command `$ convert` conflics with `C:\windows\system32\conver.exe` so to run image magic you must now use `$ magick convert <args>`.

### CODE REVIEW

**Comment by dwmkerr** [comment]
Hi @RaschidJFR it looks like the tests fail here. There is actually already a fix in place for the Windows `convert` issue:

https://github.com/dwmkerr/node-imagemagick-cli/blob/master/src/pick-cli.js#L11

If you see here, what the code will try and do is prefer a `convert` executable in a folder which has `ImageMagick` in the name; meaning the `System32` one should not be picked. But I guess your `convert` executable is in a different location.

Can you let me know the full path of your `imagemagick` binaries?

**Comment by RaschidJFR** [comment]
Hey @dwmkerr, so I found out that the problem is that version `7.0.1-Q16` no longer includes `convert.exe` (I don't know if any other executables too, I'm just getting familiar with it) and the default installation path looks like its now `C:\\Program Files\\ImageMagick-7.0.9-Q16\\magick.exe` (at least that's where it installed on my PC with default settings). [Check this out](https://manuelmeyer.net/2016/06/where-is-imagemagicks-convert-exe/).

So I modified the code to do `where magick` if the previous `where <some-cli>` returns a path wihout `ImageMagic/` in it.

The code is now working and the tests also work when not using `sandbox.stub` (it means the tests will work on an actual windows PC with an actual ImageMagic installation). Again, i'm not very familiar with mocha so I just commented one of the tests, maybe you could have a look a it.

File: src/exec.spec.js [hunk 1]
```diff
@@ -52,20 +53,58 @@ describe('exec', () => {
     //  Fake the win32 platform.
     Object.defineProperty(process, 'platform', { value: 'win32' });
 
+    // //  We expect 'childProcess.exec' to be called with 'which'.
+    // sandbox.stub(childProcess, 'exec')
```
Comment (RaschidJFR) [inline src/exec.spec.js:57]: A hand here please!
If I uncomment this block this test fails and makes the next one fail too.

File: src/exec.spec.js [hunk 2]
```diff
@@ -52,20 +53,58 @@ describe('exec', () => {
...
+
+    //  We should prefer the ImageMagick path.
+    return exec('convert -v')
+      .then(({ stdout }) => {
+        expect(stdout).to.match(/Version: ImageMagick/);
+      })
+      .catch((err) => { throw new Error(`'${err}' should not be thrown.`); });
+  });
+
+  it('can pick a CLI when ImageMagick does not have convert.exe anymore (v7.0.1-Q16)', () => {
```
Comment (RaschidJFR) [inline src/exec.spec.js:82]: I added this other test for `v7.0.1-Q1`

**Comment by RaschidJFR** [comment]
Hey @dwmkerr did you have a chance to look at the new-version problem?

> In the newer versions, the executable “convert.exe” is missing. its functionality has been merged into “magick.exe”.

That's what I'm addressing here.

**Comment by dwmkerr** [comment]
Hey @RaschidJFR, struggling to get this working on my side too! I've pushed the work I've done tonight, will keep on it but might have to wait for another evening to try and get that test working. At least only one fails now, even if they are all enabled

**Comment by RaschidJFR** [comment]
Ok, great. Sorry I'm not very useful on the mocha tests. Let me know if there's anything else I can help you with.

**Pull request merged**